test/Microsoft.AspNet.Mvc.ModelBinding.Test/Validation/ModelStateDictionaryTest.cs

```diff
@@ -733,6 +734,86 @@ public void ModelStateDictionary_NoErrorMessage_ForNonFormatException()
             Assert.Empty(error.ErrorMessage);
         }
 
+        [Fact]
+        public void ModelStateDictionary_ClearEntriesThatMatchWithKey_NonEmptyKey()
+        {
+            var dictionary = new ModelStateDictionary();
+
+            dictionary["Property1"] = new ModelState { ValidationState = ModelValidationState.Valid };
+
+            dictionary["Property2"] = new ModelState { ValidationState = ModelValidationState.Invalid };
```

Comment:
Lets talk about this one. Currently I don't look for Skipped property. I just `Unvalidate` a property and clear model state errors